File: Answer/2078.两栋颜色不同且距离最远的房子.cs
/*
 * @lc app=leetcode.cn id=2078 lang=csharp
 *
 * [2078] 两栋颜色不同且距离最远的房子
 */

// @lc code=start
public class Solution {
    public int MaxDistance(int[] colors) {
        int n = colors.Length;
        int left = 0;
        while (left < n && colors[left] == colors[n - 1]) {
            left++;
        }
        if (left == n) {
            return 0;
        }
        int right = n - 1;
        while (right >= 0 && colors[right] == colors[0]) {
            right--;
        }
        return Math.Max(n - 1 - left, right);
    }
}
// @lc code=end

File: Answer/2078.两栋颜色不同且距离最远的房子.py
#
# @lc app=leetcode.cn id=2078 lang=python3
#
# [2078] 两栋颜色不同且距离最远的房子
#

# @lc code=start
from typing import List


class Solution:
    def maxDistance(self, colors: List[int]) -> int:
        n = len(colors)
        left = 0
        while left < n and colors[left] == colors[-1]:
            left += 1
        if left == n:
            return 0
        right = n - 1
        while right >= 0 and colors[right] == colors[0]:
            right -= 1
        return max(n - 1 - left, right)

# @lc code=end

